pkg/k8s/client.go

```diff
@@ -363,8 +368,27 @@ func ServiceToMeshServices(c Controller, svc corev1.Service) []service.MeshServi
 		} else {
 			log.Warn().Msgf("k8s service %s/%s does not have endpoints but is being represented as a MeshService", svc.Namespace, svc.Name)
 		}
-		meshServices = append(meshServices, meshSvc)
+
+		if headless && endpoints != nil {
```

Comment:
Creating a helper for this is non-trivial since both this implementation and the verifier rely on two different data structures for querying: the former uses the `Controller` and the latter uses a base kubeClient. I'll whip something up though and you tell me if you think it's worth it